pandas/tests/frame/test_analytics.py

```diff
@@ -1964,6 +1964,13 @@ def test_clip_against_frame(self, axis):
         tm.assert_frame_equal(clipped_df[lb_mask], lb[lb_mask])
         tm.assert_frame_equal(clipped_df[ub_mask], ub[ub_mask])
         tm.assert_frame_equal(clipped_df[mask], df[mask])
+         # GH 20911, clipping now preserves types       
+        df1 = DataFrame(np.random.randn(1000,4), columns=['A', 'B','C','D'])
+        df2 = DataFrame(np.random.randn(1000,4), columns=['D', 'A','B','C'])
+        
+        res1 = df1.clip(lower=0,upper=df2)
```

Comment:
use result= and expected=

Reply:
will do pardon me for such a rookie mistake....